sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -283,4 +284,68 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
   async getOfferingById(offeringId: number): Promise<EducationProgramOffering> {
     return this.repo.findOne(offeringId);
   }
+
+  /**
+   * Get programs for a particular institution in paginated.
+   * @param institutionId id of the institution.
+   * @param pageSize is the number of rows shown in the table
+   * @param skip is the number of rows that is skipped/offset from the total list.
+   * For example page 2 the skip would be 10 when we select 10 rows per page.
+   * @param sortColumn the sorting column.
+   * @param sortOrder sorting order default is descending.
+   * @param searchProgramName Search the program name in the query
+   * @returns programs, locations and offerings count, programs count under the specified institution.
+   */
+  async getPaginatedProgramsForInstitution(
+    institutionId: number,
+    pageSize?: number,
+    page?: number,
+    sortColumn?: string,
+    sortOrder?: SortDBOrder,
+    searchProgramName?: string,
+  ): Promise<ProgramsOfferingSummaryPaginated> {
+    let sortByColumn = "programs.createdAt"; //Default sort column
+    const paginatedProgramQuery = await this.repo
+      .createQueryBuilder("offerings")
+      .select("programs.id", "programId")
+      .addSelect("programs.name", "programName")
+      .addSelect("programs.createdAt", "submittedDate")
+      .addSelect("locations.name", "locationName")
+      .addSelect("programs.approvalStatus", "programStatus")
+      .addSelect("COUNT(offerings.id)", "offeringsCount")
+      .innerJoin("offerings.educationProgram", "programs")
+      .innerJoin("offerings.institutionLocation", "locations")
+      .where("programs.institution.id = :institutionId", { institutionId });
+    if (searchProgramName) {
+      paginatedProgramQuery.andWhere("programs.name Ilike :searchProgramName", {
+        searchProgramName: `%${searchProgramName}%`,
+      });
+    }
+    paginatedProgramQuery
+      .groupBy("programs.id")
+      .addGroupBy("programs.name")
+      .addGroupBy("programs.createdAt")
+      .addGroupBy("locations.name")
+      .addGroupBy("programs.approvalStatus");
+    const programsCountQuery = paginatedProgramQuery.getRawMany();
+    if (pageSize) {
+      paginatedProgramQuery.limit(pageSize);
+    }
+    if (page) {
+      paginatedProgramQuery.offset(page * pageSize);
+    } else {
+      paginatedProgramQuery.offset(0);
+    }
+    if (sortColumn === "submittedDate") {
+      sortByColumn = "programs.createdAt";
```

Comment:
sortByColumn has only one value which is programs.createdAt. It is not assigned with other values from sortColumn.

Reply:
Thanks for making the change.